File: backend/internal/domain/evidence.go
package domain

import (
	"time"
)

// SourceType represents the type of source
type SourceType string

const (
	SourceTypePeerReviewedJournal SourceType = "peer_reviewed_journal"
	SourceTypePreprint            SourceType = "preprint"
	SourceTypeNews                SourceType = "news"
	SourceTypeBlog                SourceType = "blog"
	SourceTypeGovernment          SourceType = "government"
	SourceTypeUnknown             SourceType = "unknown"
)

// Evidence represents supporting or contradicting evidence for a claim
type Evidence struct {
	ID            string            `json:"id"`
	Source        string            `json:"source"`         // e.g., "Semantic Scholar"
	Title         string            `json:"title"`
	Authors       []string          `json:"authors,omitempty"`
	URL           string            `json:"url"`
	Snippet       string            `json:"snippet"`
	Credibility   float64           `json:"credibility"`   // 0.0-1.0
	Relevance     float64           `json:"relevance"`     // 0.0-1.0
	PublishedDate *time.Time        `json:"published_date,omitempty"`
	SourceType    SourceType        `json:"source_type"`
	CitationCount int               `json:"citation_count,omitempty"`
	Metadata      map[string]string `json:"metadata,omitempty"`
	CreatedAt     time.Time         `json:"created_at"`
}

File: frontend/flutter_app/lib/models/evidence.dart
import 'package:json_annotation/json_annotation.dart';

part 'evidence.g.dart';

@JsonSerializable()
class Evidence {
  final String id;
  final String source;
  final String title;
  final List<String> authors;
  final String url;
  final String snippet;
  final double credibility;
  final double relevance;

  @JsonKey(name: 'published_date')
  final String? publishedDate;

  @JsonKey(name: 'source_type')
  final String sourceType;

  @JsonKey(name: 'citation_count')
  final int? citationCount;

  final Map<String, dynamic>? metadata;

  Evidence({
    required this.id,
    required this.source,
    required this.title,
    required this.authors,
    required this.url,
    required this.snippet,
    required this.credibility,
    required this.relevance,
    this.publishedDate,
    required this.sourceType,
    this.citationCount,
    this.metadata,
  });

  factory Evidence.fromJson(Map<String, dynamic> json) =>
      _$EvidenceFromJson(json);

  Map<String, dynamic> toJson() => _$EvidenceToJson(this);

  // Helper methods
  String get sourceTypeDisplayName {
    switch (sourceType) {
      case 'peer_reviewed_journal':
        return '査読論文';
      case 'preprint':
        return 'プレプリント';
      case 'government':
        return '政府機関';
      case 'news':
        return 'ニュース';
      case 'blog':
        return 'ブログ';
      default:
        return '不明';
    }
  }

  String get credibilityLevel {
    if (credibility >= 0.9) return '非常に高い';
    if (credibility >= 0.7) return '高い';
    if (credibility >= 0.5) return '中程度';
    if (credibility >= 0.3) return '低い';
    return '非常に低い';
  }
}
